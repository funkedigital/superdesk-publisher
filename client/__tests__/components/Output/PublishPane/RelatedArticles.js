import React from "react";
import RelatedArticles from "../../../../components/Output/PublishPane/RelatedArticles";
import { render } from "@testing-library/react";
import Store from "../../../../components/Output/Store";

const related_items = [
  { title: 'relatedArticle1', tenants: [{ code: 'tenant1', name: 'tenant1' }] },
  { title: 'relatedArticle2', tenants: [{ code: 'tenant2', name: 'tenant2' }] },
];


const destinations = [
  { status: "published", tenant: { code: 'tenant1', name: 'tenant1' } },
  { status: "published", tenant: { code: 'tenant2', name: 'tenant2' } }
];

describe("Output/PublishPane/RelatedArticles", () => {

  it("renders correctly", () => {
    const { container } = render(
      <Store.Provider
        value={{
          publisher: { queryRelatedArticlesStatus: id => related_items },
          selectedItemId: 1
        }}
      >
        <RelatedArticles destinations={destinations} />
      </Store.Provider>
    );
    expect(container.firstChild).toMatchSnapshot();

  });

});
