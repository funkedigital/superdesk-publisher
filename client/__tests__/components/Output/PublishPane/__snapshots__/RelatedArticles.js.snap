// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output/PublishPane/RelatedArticles renders correctly 1`] = `
<div
  class="side-panel__content-block"
>
  <div
    class="toggle-box toggle-box--dark sp--dark-ui"
  >
    <div
      class="toggle-box__header"
    >
      <div
        class="toggle-box__chevron"
      >
        <i
          class="icon-chevron-right-thin"
        />
      </div>
      <div
        class="toggle-box__label"
      >
        Related Articles
      </div>
      <div
        class="toggle-box__line"
      />
    </div>
    <div
      class="toggle-box__content-wraper"
    >
      <div
        class="toggle-box__content"
      >
        <div
          class="spinner"
        >
          <div
            class="bounce1"
          />
          <div
            class="bounce2"
          />
          <div
            class="bounce3"
          />
        </div>
        <ul
          class="simple-list simple-list--dotted simple-list--no-padding"
        />
      </div>
    </div>
  </div>
</div>
`;
